x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/ESMessageHelper.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.security.transport;
+
+import org.elasticsearch.common.bytes.BytesReference;
+
+public class ESMessageHelper {
+
+    public static boolean isPlaintextElasticsearchMessage(BytesReference reference) {
```

Comment:
can we just put this method into `DualStackHandler` or maybe `TCPTransport` for now?